baseplate/experiments/__init__.py

```diff
@@ -134,25 +138,43 @@ def variant(self, name, user=None, bucketing_event_override=None,
         :rtype: :py:class:`str`
         :return: Variant name if a variant is active, None otherwise.
         """
+
         experiment = self._get_experiment(name)
 
         if experiment is None:
             return None
 
         inputs = dict(kwargs)
+
+        """ clean up user id. If not a baseplate user, we risk inconsistency."""
         if user:
             if user.is_logged_in:
                 inputs["user_id"] = user.id
                 inputs["user_roles"] = user.roles
+                inputs["logged_in"] = user.is_logged_in
             else:
                 inputs["user_id"] = user.loid
-            inputs["logged_in"] = user.is_logged_in
+                inputs["logged_in"] = False
+
+        #loid is numeric. Convert it to a base-36 string; Ideally, this will
+        #   be fixed before the caller sends it in, but this will catch
+        if isinstance(inputs.get("user_id"), int):
+            if (inputs.get("user_id") < 0):
+                logger.warning("unable to bucket user with invalid user_id < 0")
+                return None
+            inputs["user_id"] = to36(inputs["user_id"])
+
+        # if it's not already a fullname, let's make it one (and strip leading
+        #    zeroes while we're at it. And lowercase it to be safe.)
+        if inputs.get("user_id") and not inputs["user_id"].startswith("t2_"):
+            inputs["user_id"] = "t2_" + inputs["user_id"].lstrip("0").lower()
```

Comment:
I don't think we want to fix this, I think we want to raise an error if someone gives us the wrong format if we want to be pedantic.

Reply:
fair point.